web_poet/rules.py

```diff
@@ -259,6 +339,48 @@ def search_overrides(self, **kwargs) -> List[ApplyRule]:
         warnings.warn(msg, DeprecationWarning, stacklevel=2)
         return self.search(**kwargs)
 
+    def _match_url_for_page_object(
+        self, url: Union[_Url, str], matcher: Optional[URLMatcher] = None
+    ) -> Optional[Type[ItemPage]]:
+        """Returns the page object to use based on the URL and URLMatcher."""
+        if not url or matcher is None:
+            return None
+
+        rule_id = matcher.match(str(url))
+        if rule_id is not None:
+            return self._rules[rule_id].use
+
+    def overrides_for(
+        self, url: Union[_Url, str]
+    ) -> Mapping[Type[ItemPage], Type[ItemPage]]:
+        """Finds all of the page objects associated with the given URL and
+        returns a Mapping where the 'key' represents the page object that is
+        **overridden** by the page object in 'value'.
+
+        See example: :ref:`rules-overrides_for-example`.
+        """
+        result: Dict[Type[ItemPage], Type[ItemPage]] = {}
+        for target, matcher in self._overrides_matchers.items():
+            if target is None:
+                continue
+            po = self._match_url_for_page_object(url, matcher)
+            if po:
+                result[target] = po
+        return result
```

Comment:
What do you think about something like this?

```suggestion
        result: Dict[Type[ItemPage], Type[ItemPage]] = {}
        for replaced_page, matcher in self._overrides_matchers.items():
            if replaced_page is None:
                continue
            page = self._match_url_for_page_object(url, matcher)
            if page:
                result[replaced_page] = page
        return result
```